components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java

```diff
@@ -207,15 +250,18 @@ private StorageTx openStorageTx(final ODatabaseDocumentTx db) {
     BlobStore blobStore = blobStoreManager.get(config.blobStoreName);
     return StateGuardAspect.around(
         new StorageTxImpl(
```

Comment:
Looking at the long list of parameters here I'm thinking we should look at the factory pattern or even assisted-inject to help manage this.